ibm/service/vpc/resource_ibm_is_subnet_public_gateway_attachment.go

```diff
@@ -0,0 +1,332 @@
+			},
+		},
+	}
+}
+
+func resourceIBMISSubnetPublicGatewayAttachmentCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	subnet := d.Get(isSubnetID).(string)
+	publicGateway := d.Get(isPublicGatewayID).(string)
+
+	publicGatewayIdentity := &vpcv1.PublicGatewayIdentity{
+		ID: &publicGateway,
+	}
+
+	// Construct an instance of the SetSubnetPublicGatewayOptions
+	setSubnetPublicGatewayOptions := &vpcv1.SetSubnetPublicGatewayOptions{
+		ID:                    &subnet,
+		PublicGatewayIdentity: publicGatewayIdentity,
+	}
+
+	pg, response, err := sess.SetSubnetPublicGatewayWithContext(context, setSubnetPublicGatewayOptions)
+
+	if err != nil {
+		log.Printf("[DEBUG] Error while attaching public gateway(%s) to subnet(%s) %s\n%s", publicGateway, subnet, err, response)
+		return diag.FromErr(fmt.Errorf("[ERROR] Error while attaching public gateway(%s) to subnet(%s) %s\n%s", publicGateway, subnet, err, response))
+	}
+	d.SetId(subnet)
+	_, err = isWaitForSubnetPublicGatewayAvailable(sess, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[INFO] Public Gateway : %s", *pg.ID)
+	log.Printf("[INFO] Subnet ID : %s", subnet)
+
+	return resourceIBMISSubnetPublicGatewayAttachmentRead(context, d, meta)
+}
+
+func resourceIBMISSubnetPublicGatewayAttachmentRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	id := d.Id()
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getSubnetPublicGatewayOptionsModel := &vpcv1.GetSubnetPublicGatewayOptions{
+		ID: &id,
+	}
+	pg, response, err := sess.GetSubnetPublicGatewayWithContext(context, getSubnetPublicGatewayOptionsModel)
+
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("[ERROR] Error getting subnet's (%s) attached public gateway: %s\n%s", id, err, response))
+	}
+	d.Set(isPublicGatewayName, pg.Name)
+	if pg.FloatingIP != nil {
+		floatIP := map[string]interface{}{
+			"id":                             *pg.FloatingIP.ID,
+			isPublicGatewayFloatingIPAddress: *pg.FloatingIP.Address,
+		}
+		d.Set(isPublicGatewayFloatingIP, floatIP)
+	}
+	d.Set(isPublicGatewayStatus, pg.Status)
+	if pg.ResourceGroup != nil {
+		d.Set(isPublicGatewayResourceGroup, *pg.ResourceGroup.ID)
+		d.Set(flex.ResourceGroupName, *pg.ResourceGroup.Name)
+	}
+	d.Set(isPublicGatewayVPC, *pg.VPC.ID)
+	d.Set(isPublicGatewayZone, *pg.Zone.Name)
+	d.Set(IsPublicGatewayResourceType, pg.ResourceType)
+	d.Set(isPublicGatewayCRN, pg.CRN)
+
+	return nil
+}
+
+func resourceIBMISSubnetPublicGatewayAttachmentUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if d.HasChange(isPublicGatewayID) {
```

Comment:
should the change of isPublicGatewayID force new ?

Reply:
all fields will be forcenew/computed then.